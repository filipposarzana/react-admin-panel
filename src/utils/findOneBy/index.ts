import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
