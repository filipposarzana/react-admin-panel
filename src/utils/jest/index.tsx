import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
