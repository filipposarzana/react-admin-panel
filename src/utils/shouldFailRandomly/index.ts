export const shouldFailRandomly = () => Math.round(Math.random()) === 0
