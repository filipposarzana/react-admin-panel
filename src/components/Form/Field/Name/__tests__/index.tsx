import { renderer } from '~/utils/jest'
import { Name } from '..'

describe('Name', () => {
  it('matches snapshot without defaultValue', () => {
    const { output } = renderer.shallow(<Name />)

    expect(output).toMatchSnapshot()
  })

  it('matches snapshot with defaultValue', () => {
    const { output } = renderer.shallow(<Name defaultValue="Name" />)

    expect(output).toMatchSnapshot()
  })
})
