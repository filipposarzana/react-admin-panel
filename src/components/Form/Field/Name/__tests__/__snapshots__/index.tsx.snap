// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name matches snapshot with defaultValue 1`] = `
<styled.div>
  <FormLabel
    htmlFor="name"
    label="Name"
  />
  <Controller
    defaultValue="Name"
    name="name"
    render={[Function]}
    rules={
      Object {
        "required": "Mandatory",
      }
    }
  />
  <FormError />
</styled.div>
`;

exports[`Name matches snapshot without defaultValue 1`] = `
<styled.div>
  <FormLabel
    htmlFor="name"
    label="Name"
  />
  <Controller
    defaultValue=""
    name="name"
    render={[Function]}
    rules={
      Object {
        "required": "Mandatory",
      }
    }
  />
  <FormError />
</styled.div>
`;
